docs/DeveloperGuide.md

```diff
@@ -1,15 +1,93 @@
 # Developer Guide
+Welcome to the NUSPlanner Developer Guide!
+We hope this documents serves useful to understand the behind-the-scenes working of our product. 
+
+## Table of Contents
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}'
+
+## Getting started
+
+Refer to the user guide [here](https://github.com/AY2223S2-CS2113-F13-3/tp/blob/master/docs/UserGuide.md).
+
 
 ## Design & implementation
```

Comment:
I like how all the diagrams are in details and that's a whole lotta effort to do it! However it would be greater if you can have an overall architecture!